cmd/lifecycle/exporter.go

```diff
@@ -313,13 +363,13 @@ func (ea exportArgs) initRemoteAppImage(analyzedMD platform.AnalyzedMetadata) (i
 
 	if analyzedMD.Image != nil {
 		cmd.DefaultLogger.Infof("Reusing layers from image '%s'", analyzedMD.Image.Reference)
-		ref, err := name.ParseReference(analyzedMD.Image.Reference, name.WeakValidation)
+		// ensure previous image is on same registry as output image
+		analyzedRegistry, err := parseRegistry(analyzedMD.Image.Reference)
 		if err != nil {
 			return nil, "", cmd.FailErr(err, "parse analyzed registry")
 		}
-		analyzedRegistry := ref.Context().RegistryStr()
-		if analyzedRegistry != ea.registry {
-			return nil, "", fmt.Errorf("analyzed image is on a different registry %s from the exported image %s", analyzedRegistry, ea.registry)
+		if analyzedRegistry != ea.targetRegistry {
```

Comment:
I know this wasn't a change, but why does this case produce a hard failure?

Reply:
I **think** because of layer re-use. Though I seem to recall @ekcasey mentioned we might be able to copy the layers in the future.